api/api_sources/sources/database/factory/helper.ts

```diff
@@ -326,6 +329,11 @@ export function Destroyer(controller: DataController) {
 
 export function ModelFactory(controller: DataController) {
     return async (options: FactoryOptions = { schemaChain: []}, inputData?: any): Promise<any> => {
+        const schema: BaseSchema = controller.schemaObject;
+        if (schema.hasDefaultValues) {
+            const obj = inputData !== undefined && typeof inputData === 'number' ? await controller.findById(inputData || 1) : await controller.random();
```

Comment:
Just curious! How does the precedence work here? Does the `typeof` always have more precedence than `===`? So, that it runs the check between `typeof inputData` and `number` and not trying to find the value for, typeof `inputData === number` ?

Reply:
1. Mostly precedence of unary operators higher than any binary operator or ternary.
2. typeof inputData returns type as string which is compared with 'number'
3. I used ternary operator here  expression ? true value : false value ... similar to if else 
which has minimum precedence

Reply:
Thanks for the clarification 👍